<launch>
    <!-- Set parameters -->
    <rosparam file="$(find geolocation)/config/ros_params.yaml" />

    <!-- Set topic names -->
    <rosparam file="$(find geolocation)/config/topic_names.yaml" />

    <!-- Launch tf2 broadcasters -->
    <node name="base_tf2" pkg="geolocation" type="base_tf2_broadcaster" />
    <node name="map_tf2" pkg="geolocation" type="map_tf2_broadcaster" />

    <!-- Launch geolocation script -->
    <node name="geolocation" pkg="geolocation" type="geo_publisher.py" args="0" output="screen" />

    <!-- Launch dummy GPS publisher (if enabled) -->
    <arg name="dummy_gnss" default="false" />

    <group if="$(arg dummy_gnss)">
        <rosparam file="$(find geolocation)/config/dummy_params.yaml" />
        <node name="dummy_brng" pkg="geolocation" type="dummy_bearing_pub" />
        <node name="dummy_elv" pkg="geolocation" type="dummy_elv_pub" />
        <node name="dummy_gnss" pkg="geolocation" type="dummy_gnss_pub" />
    </group>

    <!-- Launch bag file -->
    <node name="bag" pkg="rosbag" type="play" args="-s 20 -u 25 $(find geolocation)/launch/rosbag-3.bag" />

    <!-- Launch rviz with the preconfiguration -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find geolocation)/launch/bag.rviz" />
</launch>